packages/graphql-transformer-core/src/util/sanity-check.ts

```diff
@@ -212,6 +212,50 @@ export function cantAddAndRemoveGSIAtSameTime(diff: Diff, currentBuild: Diffable
   }
 }
 
+/**
+ * Throws a helpful error when a customer is trying to complete an invalid migration.
+ * Users are unable to add multiple GSIs at the same time in update stage.
+ * @param diffs The set of diffs between currentBuild and nextBuild.
+ * @param currentBuild The last deployed build.
+ * @param nextBuild The next build.
+ */
+export function cantMutateMultipleGSIAtUpdateTime(diffs: Diff[], currentBuild: DiffableProject, nextBuild: DiffableProject) {
+  function throwError(stackName: string, tableName: string) {
+    throw new InvalidMigrationError(
+      `Attempting to mutate more than 1 global secondary index at the same time on the ${tableName} table in the ${stackName} stack. `,
+      'You may only mutate one global secondary index in a single CloudFormation stack update. ',
+      'If using @key, include one @key at a time. ' +
+        'If using @connection, just add one new @connection which is using @key, run `amplify push`, ',
+    );
+  }
+
+  if (diffs) {
+    let gsiCountAdded: number = 0; // max gsiCountAdded = 1 // for update flow
+    let gsiCountRemoved: number = 0; // max gsiCountRemoved = 1 // for update flow
+    for (const diff of diffs) {
+      if (
+        // implies a field was changed in a GSI after it was created.
+        // Path like:["stacks","Todo.json","Resources","TodoTable","Properties","GlobalSecondaryIndexes", ... ]
+        diff.kind === 'A' &&
+        diff.path.length >= 6 &&
+        diff.path[5] === 'GlobalSecondaryIndexes'
+      ) {
+        if (diff.item.kind === 'N') {
+          gsiCountAdded += 1;
+        }
+        if (diff.item.kind === 'D') {
+          gsiCountRemoved += 1;
+        }
+        if (gsiCountAdded > 1 || gsiCountRemoved > 1) {
```

Comment:
I think there is also a  case where one can't add and delete an GSI in the same update. This does not seem to handle that case

Reply:
I have added the case for update and delete GSI at same time